### DISCUSSION TITLE
Getting the features repo off the ground

### DISCUSSION THREAD

**Discussion by jcbhmr** [devcontainers-community/features#11] [General]
If/when https://github.com/devcontainers-community/templates/pull/5 is merged, I think getting a first feature is a good next step. Thoughts? Should we stick to just templates for now since, as @eitsupi said: they are harder

> Note that Templates and Features have a very different level of maintenance importance.
That is, Templates are only downloaded once, while Features are expected to continue to be downloaded many times in the future.
>
> So maintenance of Features is far more important than maintenance of Templates, and when Features are handled by such an organization, the maintainer needs to be clearly stated, I think.

_in my opinion_ I think getting a foot into both templates and features is a good way to get started.

But then again, maybe we should take it slow and focus on just the jupyter template stuff for a few more days/week (🥵 we don't want to overwork @jcbhmr and @eitsupi) before tackling anything else. Thoughts?

/cc @eitsupi

**Comment by eitsupi** [comment]
Is #5 perhaps #5 from another repository?

> _in my opinion_ I think getting a foot into both templates and features is a good way to get started.
> 
> But then again, maybe we should take it slow and focus on just the jupyter template stuff for a few more days/week (🥵 we don't want to overwork @jcbhmr and @eitsupi) before tackling anything else. Thoughts?

I think we can start with some transfers of templates from https://github.com/microsoft/vscode-dev-containers for some reasons.

- These have been in use for a while, but are now deprecated and not under maintenance.
- The scope is clear. (No possibility of unlimited increase)
- Maintenance of Templates are considered less effort than maintenance of Features. (It is the base image and the Features that are more difficult to maintain!)

Again, both Features and Templates can grow without limit unless the scope is clearly defined, and there must be clear criteria for what is to be added.

**Reply by jcbhmr** [reply]
> Is #5 perhaps #5 from another repository?

Yeah lmao it was supposed to link https://github.com/devcontainers-community/templates/pull/5 🤣

> Again, both Features and Templates can grow without limit unless the scope is clearly defined, and there must be clear criteria for what is to be added.

You're right to think of ballooning responsibility. I think I'm being a bit overzealous. Thanks for keeping your eye on scope. 👍 I agree with your idea to stick with this:

> I think we can start with some transfers of templates from https://github.com/microsoft/vscode-dev-containers

For right now, I agree that my proposal in this discussion was a bit _too much_ and this is a better path forward for now. Thanks for moderating my enthusiasm! ❤

**Reply by eitsupi** [reply]
Your passion is really great!
However, I think we need to explore the state of sustainability to make it easier for contributors to participate.

**Reply by jcbhmr** [reply]
> However, I think we need to explore the state of sustainability to make it easier for contributors to participate.

Totally key. You're right! Sticking with just that https://github.com/devcontainers-community/templates/pull/5 template and making it and the docs _really good_ seems like a good idea.

**Reply by eitsupi** [reply]
For example, as shown below, a large amount of PR can be left unattended and not good for both parties if acceptance criteria are not indicated.......
https://github.com/microsoft/vscode-dev-containers/pulls

After looking at this repository for several years, I felt it was really important to have a scope, guidelines, CI, and other standards in place.
